ast/stats.go

```diff
@@ -39,7 +39,10 @@ type AnalyzeTableStmt struct {
 	Incremental bool
 	// HistogramOperation is set in "ANALYZE TABLE ... UPDATE/DROP HISTOGRAM ..." statement.
 	HistogramOperation HistogramOperationType
-	ColumnNames        []*ColumnName
+	// ColumnNames indicate the columns whose statistics need to be collected.
+	ColumnNames []*ColumnName
+	// PredicateColumns is true when we only collect statistics of predicate columns and indexed columns.
+	PredicateColumns bool
```

Comment:
What is "predicate columns"?

Reply:
Predicate columns are the columns that has been used in a query as a part of a filter, join condition, or group by clause. Currently, we collect the statistics of all columns when executing ANALYZE, which is kind of waste since not all column statistics will be used. Actually, only the statistics of predicate columns are helpful to cost based optimization. When `PredicateColumns` is true, ANALYZE only collects the statistics of predicate columns so the cost(time, memory, cpu ...) of ANALYZE can be reduced. https://docs.google.com/document/d/12_SiDLInsQyHj915U4Np4uphpH3sFhQk0KzMKRAmyi4/edit?usp=sharing has a detailed description. Redshift's ANALYZE also has PREDICATE COLUMNS option, see  https://docs.aws.amazon.com/redshift/latest/dg/t_Analyzing_tables.html and https://docs.aws.amazon.com/redshift/latest/dg/r_ANALYZE.html.